browser/brave_shields/brave_shields_util_profiles_unittest.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_shields/content/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/cookie_settings_factory.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "chrome/test/base/testing_profile.h"
+#include "chrome/test/base/testing_profile_manager.h"
+#include "components/content_settings/core/browser/cookie_settings.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace {
+
+using ScopedIncognitoProfile =
+    std::unique_ptr<Profile, decltype([](Profile* p) {
+                      ASSERT_TRUE(!!p->IsOffTheRecord());
+                      p->GetOriginalProfile()->DestroyOffTheRecordProfile(p);
+                    })>;
+
+testing::Message& operator<<(testing::Message& m,
+                             brave_shields::ControlType c) {
+  return m << brave_shields::ControlTypeToString(c);
+}
+
+}  // namespace
+
+namespace brave_shields {
+
+class BraveShieldsUtilProfilesTest : public testing::Test {
+ public:
+  BraveShieldsUtilProfilesTest()
+      : local_state_(TestingBrowserProcess::GetGlobal()) {}
+  ~BraveShieldsUtilProfilesTest() override = default;
+
+  TestingProfile* regular_profile() { return &profile_; }
+  ScopedIncognitoProfile incognito_profile() {
+    return ScopedIncognitoProfile(regular_profile()->GetOffTheRecordProfile(
+        Profile::OTRProfileID::PrimaryID(),
+        /*create_if_needed=*/true));
+  }
+
+  HostContentSettingsMap* hcsm(Profile* profile) {
+    return HostContentSettingsMapFactory::GetForProfile(profile);
+  }
+
+  HostContentSettingsMap* hcsm(ScopedIncognitoProfile& profile) {
+    return hcsm(profile.get());
+  }
```

Comment:
nit: I feel these functions are making the code harder to read, rather than just using `HostContentSettingsMapFactory::GetForProfile` in the body of `RunTest`, but I leave it to you to decide which approach is better.

I think the naming here can be improved though. If you look in upstream they name things fully for these cases (e.g. `GetControlledFrameDefaultContentSetting`).